numpy/ma/core.py

```diff
@@ -5439,7 +5439,7 @@ def argsort(self, axis=np._NoValue, kind=None, order=None,
             When the array contains unmasked values at the same extremes of the
             datatype, the ordering of these values and the masked values is
             undefined.
-        fill_value : {var}, optional
+        fill_value : scalar or None, optional
```

Comment:
Is the mention of `None` not redundant here?
Considering the parameter is already specified as `optional`.

Reply:
I was unsure; but the description for optional says:

> If it is not necessary to specify a keyword argument, use optional:

So I interpreted the structure before the comma as "values the can be used for fill_value", and `.argsort(..., fill_value=None)` is definitely a valid calling of this methond.

I agree that in practice `, optional` seem to be equivalent to `or None`, though it is far from the only place that use `or None, optional`:

```
numpy/core/fromnumeric.py
690:    axis : int or None, optional
782:    axis : int or None, optional
853:    axis : int or None, optional
1017:    axis : int or None, optional

numpy/core/multiarray.py
1208:    count : int or None, optional

numpy/distutils/ccompiler_opt.py
149:        "autovec": bool or None, optional
1176:        'names': sequence or None, optional
1181:        'force_flags': list or None, optional
1441:        'force_flags': list or None, optional
1475:        'force_flags': list or None, optional
```

among others...

Unless I misunderstood what you asked ...

Reply:
Strictly speaking, I think it would be `{scalar, None}`, but it has been pretty loose for some years now :) None is a valid value and it is last in the brackets because it is the default. That said, I've found it better to mention the default in the docstring and not rely on users parsing curly brackets, so this is fine. The problem is that None is sometimes used to signal some other default, so it needs to be explained what the default behavior is in those cases.

Reply:
whatever you prefer; let me know. And either here and everywhere.